scripts/xdist/pytest_worker_manager.py

```diff
@@ -0,0 +1,226 @@
+                        'Groups': security_group_ids,
+                        'SubnetId': subnet,
+                        },
+                    ],
+                    TagSpecifications=[
+                        {
+                            'ResourceType': 'instance',
+                            'Tags': [
+                                {"Key": "master", "Value": "build.testeng.edx.org"},
+                                {"Key": "worker", "Value": "pytest_xdist_worker"},
+                                {"Key": "Name", "Value": "xdist agent"}
+                            ]
+                        }
+                    ]
+                )
+            except ClientError as err:
+                # Handle AWS throttling with an exponential backoff
+                if retry == self.MAX_RUN_WORKER_RETRIES:
+                    raise Exception(
+                        "MAX_RUN_WORKER_RETRIES ({}) reached while spinning up workers due to AWS throttling.".format(self.MAX_RUN_WORKER_RETRIES)
+                    )
+                logger.info("Hit error: {}. Retrying".format(err))
+                countdown = 2 ** retry
+                logger.info("Sleeping for {} seconds".format(countdown))
+                time.sleep(countdown)
+            else:
+                break
+
+        for instance_response in response['Instances']:
+            worker_instance_ids.append(instance_response['InstanceId'])
+
+        # Wait for workers to finish spinning up
+        not_running = worker_instance_ids[:]
+        ip_addresses = []
+        all_running = False
+        for attempt in range(0, self.WORKER_BOOTUP_TIMEOUT_MINUTES * 30):
+            try:
+                list_workers_response = self.ec2.describe_instances(InstanceIds=not_running)
+            except:
+                logger.info("Exception hit trying to describe instances.")
+                continue
+            del not_running[:]
+            for reservations in list_workers_response['Reservations']:
+                for instance_info in reservations['Instances']:
+                    if instance_info['State']['Name'] == "running":
+                        ip_addresses.append(instance_info['PrivateIpAddress'])
+                    else:
+                        not_running.append(instance_info['InstanceId'])
+
+            if len(not_running) > 0:
+                logger.info("Still waiting on {} workers to spin up".format(len(not_running)))
+                time.sleep(5)
+            else:
+                logger.info("Finished spinning up workers")
+                all_running = True
+                break
+
+        if not all_running:
+            raise Exception(
+                "Timed out waiting to spin up all workers."
+            )
+        logger.info("Successfully booted up {} workers.".format(number_of_workers))
+
+        not_ready_ip_addresses = ip_addresses[:]
+        logger.info("Checking ssh connection to workers.")
```

Comment:
can we shift to single qoutes here, line 119, line 121 and elsewhere in this file?